krm-functions/lib/nad/v1/nad.go

```diff
@@ -99,18 +103,14 @@ func NewFromYAML(b []byte) (*NadStruct, error) {
 
 // NewFromGoStruct creates a new parser interface
 // It expects a go struct representing the interface krm resource
-func NewFromGoStruct(x *nadv1.NetworkAttachmentDefinition) (*NadStruct, error) {
-	p, err := kubeobject.NewFromGoStruct[*nadv1.NetworkAttachmentDefinition](x)
+func NewFromGoStruct(b *nadv1.NetworkAttachmentDefinition) (*NadStruct, error) {
+	p, err := kubeobject.NewFromGoStruct[nadv1.NetworkAttachmentDefinition](*b)
```

Comment:
Are you sure that this compiles with the latest nephio-project/nephio?
It should be:  `kubeobject.NewFromGoStruct[nadv1.NetworkAttachmentDefinition](b)`, or simply  `kubeobject.NewFromGoStruct(b)`.